server/routes/productRoute.js

```diff
@@ -3,32 +3,71 @@ const router = express.Router();
 const productData = require("../data/products");
 const session = require("express-session");
 
+// router.get("/", async (req, res) => {
+//   try {
+//     let prodList = await productData.getAllProducts();
+//     res.status(200).render("homePage/homePage", {
+//       products: prodList,
+//     });
+//   } catch (e) {
+//     console.log(e);
+//     res.status(404).send(e);
+//   }
+// });
+
 router.post("/search", async (req, res) => {
-    const body = req.body;
-    let option = document.getElementById("searchSelect");
-    let searchTerm = body.search;
-    if (!searchTerm || searchTerm.trim() == 0) {
-        return "Error: Search term blank";
-    } else {
-        searchTerm = searchTerm.toLowerCase();
-        if (option.value === "By Name") {
-            try {
-                let search_List = await products.getProductByProductName(
-                    body.search
-                );
-                res.status(200).render("homePage/homePage", {
-                    products: search_List,
-                });
-            } catch (e) {
-                return res.status(500).message(e.message);
-            }
-        }
-        if (option.value === "By Tag") {
-            try {
-                let search_List = await products.getProductbyTag(body.search);
-            } catch (e) {
-                return res.status(500).send();
-            }
-        }
+  const body = req.body;
+  console.log("body", body);
+
+  //console.log(option);
+  let searchTerm = body.search;
+  console.log(searchTerm);
+  if (!searchTerm || searchTerm.trim() == 0) {
+    return "Error: Search term blank";
+  } else {
+    searchTerm = searchTerm.toLowerCase();
+    console.log(searchTerm);
+    console.log(option.value);
+    if (option.value === "name") {
+      try {
+        let search_List = await productData.getProductByProductName(
+          body.search
+        );
+        //return only the json
+
+        res.status(200).render("searchPage/searchPage", {
+          products: search_List,
+        });
+      } catch (e) {
+        return res.status(404).render("errorPage/noSearch");
+      }
     }
+    if (option.value === "tag") {
+      try {
+        let search_List = await productData.getProductbyTag(body.search);
+        res.status(200).render("searchPage/searchPage", {
+          products: search_List,
+        });
+      } catch (e) {
+        return res.status(404).render("errorPage/noSearch");
+      }
+    }
+  }
+});
+
+router.post("/addProduct", async (req, res) => {
```

Comment:
I think the portion to check if user is logged in is automatically handled if you include authMiddleware

Reply:
This I had in my previous code too, This was before the addProduct was takenby Shilin. I'm not sure why this is coming up as a change, I have not started this as of yet